### PR TITLE
Fix traffic control message when starting in space

### PR DESCRIPTION
**Pull request by zonkmachine**
Fixes https://github.com/pioneerspacesim/pioneer/issues/6219 by making sure playerInControlledSpace == false when starting in space.

I don't know if there are possible edge cases where this new algorithm wouldn't work. If for instance we start in an orbit around a star but we're still in a controlled space via a space station.

### CODE REVIEW

**Comment by zonkmachine** [comment]
I've tested this some more and it seem to work fine.

**Comment by zonkmachine** [comment]
Merge?

**Comment by impaktor** [comment]
What happens if the player is spawned outside a station? They're not docked, but in controlled space? Or do we not do that?

**Comment by zonkmachine** [comment]
> What happens if the player is spawned outside a station? They're not docked, but in controlled space? Or do we not do that?

Let's find out. Starting in orbit around Phobos:
`location   = SystemPath.New(0,0,0,0,21),`

phobostrafficcontroll

In these cases, what is the expected result? I'm kind of cool with an edge case like this but it could be solved with an extra test for proximity to a station.

**Comment by zonkmachine** [comment]
I guess in the situation above 27 km from Phobos Base, you wouldn't expect to start with a message like that. I'll fix it.

**Comment by zonkmachine** [comment]
> What happens if the player is spawned outside a station? They're not docked, but in controlled space? Or do we not do that?

Fixed
```lua
	elseif Game.player.flightState == "DOCKED" or
			station:DistanceTo(Game.player) < station.lawEnforcedRange then
		playerInControlledSpace = true
```

**Comment by zonkmachine** [comment]
I've tested it some more for possible edge cases but I'm low on ideas. It now works.

_(Edit: Both on master and this PR)_ It doesn't seem to be especially defined what the behavior should be when two stations are really close to each other. You get the message from one when you enter the area and then a goodby from the other when you exit after having transited from the first one. You will not get any traffic messages when moving in between the however. This is not a great problem. You could perhaps make control zones of stations smaller when they overlap but that may be just unnecessary noise. Maybe a different message could be used. "Now leaving 'x', we hand you over to 'y'. Something like that.

**Comment by zonkmachine** [comment]
Merge?

File: data/modules/StationTrafficControl/StationTrafficControl.lua [hunk 1]
```diff
@@ -106,8 +109,12 @@ end
 local loaded_data
 
 local onGameStart = function ()
+	local station = Game.player:FindNearestTo("SPACESTATION")
```
Comment (impaktor) [inline data/modules/StationTrafficControl/StationTrafficControl.lua]: I assume this will never be nil, as can happen if the player is in hyperspace, but since we can't start the game in hyperspace, this should be fine, I assume.

**Review by impaktor — APPROVED** [review]
I've not tested it, but looks fine when I ran the proposed changes in the software on my wetware (brain), it did not explode.

File: data/modules/StationTrafficControl/StationTrafficControl.lua (hunk 1 above)
Comment (zonkmachine) [reply data/modules/StationTrafficControl/StationTrafficControl.lua]: Well spotted. No, you just need to start in a system with no station and this code will crash. Sorry about that.

Comment (zonkmachine) [reply data/modules/StationTrafficControl/StationTrafficControl.lua]: Fix pushed

File: data/modules/StationTrafficControl/StationTrafficControl.lua [hunk 2]
```diff
@@ -106,8 +109,12 @@ end
...
 	if (loaded_data) then
 		playerInControlledSpace = loaded_data.playerInControlledSpace
+	elseif Game.player.flightState == "DOCKED" or
+			station and station:DistanceTo(Game.player) < station.lawEnforcedRange then
```
Comment (zonkmachine) [inline data/modules/StationTrafficControl/StationTrafficControl.lua:116]: @impaktor I added **'station and'** here in response to your earlier comment on station being nil. I use no parenthesis here and trust in short-circuit evaluation. This is OK and clean right?

File: data/modules/StationTrafficControl/StationTrafficControl.lua (hunk 1 above)
Comment (impaktor) [reply data/modules/StationTrafficControl/StationTrafficControl.lua]: well, I anti-spotted it, thinking it was not an issue. Två kloka huvuden, or whatever the saying is.

Comment (zonkmachine) [reply data/modules/StationTrafficControl/StationTrafficControl.lua]: Merge at will...
